tianshou/data/batch.py

```diff
@@ -511,13 +530,42 @@ def stack_(self,
 
     @staticmethod
     def stack(batches: List['Batch'], axis: int = 0) -> 'Batch':
-        """Stack a :class:`~tianshou.data.Batch` object into a
-        single new batch.
+        """Stack a :class:`~tianshou.data.Batch` object into a single new
+        batch.
         """
         batch = Batch()
         batch.stack_(batches, axis)
         return batch
 
+    def empty_(self) -> 'Batch':
+        """Return an empty a :class:`~tianshou.data.Batch` object with 0 or
+        ``None`` filled.
+        """
+        for k, v in self.items():
+            if v is None:
+                continue
+            if isinstance(v, Batch):
+                self.__dict__[k].empty_()
+            elif isinstance(v, np.ndarray) and v.dtype == np.object:
+                self.__dict__[k].fill(None)
+            elif isinstance(v, torch.Tensor):  # cannot apply fill_ directly
+                self.__dict__[k] = torch.zeros_like(self.__dict__[k])
```

Comment:
You could use `fill_` for torch tensors, it is more robust (reserve dtype and device) and avoid memory allocation.